components/org.wso2.carbon.identity.mgt/src/main/java/org/wso2/carbon/identity/mgt/impl/util/builder/event/EventInterceptorTemplate.java

```diff
@@ -110,7 +110,17 @@ public T getResult() {
             eventService.pushEvent(event, messageContext);
         } catch (IdentityException e) {
             String message = String.format("Error while handling %s event.", eventId);
-            throw new IdentityStoreException(message, e);
+            IdentityStoreException identityStoreException = new IdentityStoreException(message, e);
+
+            if (e.getErrorCode() != null) {
+                try {
+                    int errorCode = Integer.parseInt(e.getErrorCode());
+                    identityStoreException.setErrorCode(errorCode);
+                } catch (NumberFormatException e1) {
+                    // Ignore error code.
```

Comment:
Ideally we need to have a default number assigned for this. Shall we create a jira for this.?